Content.Server/GameTicking/GameTicker.Spawning.cs

```diff
@@ -274,9 +275,11 @@ private List<EntityUid> GetSpawnableStations()
                     Loc.GetString("job-greet-station-name", ("stationName", metaData.EntityName)));
             }
 
+
             // Arrivals is unable to do this during spawning as no actor is attached yet.
             // We also want this message last.
-            if (!silent && lateJoin && _arrivals.Enabled)
+            // Arrival System adds a PendingClockInComponent, while the Cryosleep System does not.
+            if (!silent && lateJoin && HasComp(mob, typeof(PendingClockInComponent)))
             {
```

Comment:
1. You're using the non-generic version which will be slower because this has to check type.
2. Rather than hard-coding this in for servers that don't have it better to just raise an event / have the spawners handle it and decide if this needs to be here.

Reply:
Why does this message appear even if ArrivalsSystem is disabled? This message should be somewhere inside this system, not in the player's spawn code